### ISSUE TITLE
Create a request

### ISSUE THREAD

**Issue by lmullane** [bcgov/foi-flow#30]
As an intake analyst, I want to be able to create a new request, so that I can add paper/email requests to the system

**Acceptance Criteria**	

**Scenario 1: xxxxxx**
* GIVEN ... (you or your condition)
* WHEN ... (what you do) 
* THEN ... (what you see) 

**Scenario 2: xxxxxx**
* GIVEN ... (you or your condition)
* WHEN ... (what you do) 
* THEN ... (what you see) 

**Scenario 3: xxxxxx**
...

**Dependencies? What is the impact of this dependency? (If so, link dependency in the ticket, make it visible in a team´s backlog)**

**Validation Rules? (If yes, list here)**	

**Design**
@xxx - please link the Design here

**Definition of Ready**

1. [ ] Is there a well articulated User Story?
2. [ ] Is there Acceptance Criteria that covers all scenarios (happy/sad paths)?
3. [ ] If there is a user interface, is there a design? 
4. [ ] Does the user story need user research/validation?
5. [ ] Does this User Story needs stakeholder approval?
6. [ ] Design / Solution accepted by Product Owner
7. [ ] Is this user story small enough to be completed in a Sprint? Should it be split?
8. [ ] Are the dependencies known/ understood? (technical, business, regulatory/policy)
9. [ ] Has the story been estimated?

**Definition of Done**

1.     [ ] Passes developer unit tests
2.    [ ] Passes peer code review
3.    [ ] If there's a user interface, passes UX assurance
4.    [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
5.	[ ] Confirm Test cases built and succeeding 
6.	[ ] No regression test failures
7.	[ ] Test coverage acceptable by Product Owner
8.	[ ] Ticket ready to be merged to master or story branch
9.	[ ] Developer to list Config changes/ Update documents and designs
10.	[ ] Can be demoed in Sprint Review 
11.	[ ] Tagged as part of a Release
12.   [ ] Feature flagged if required
13.  [ ] Change Management activities done?

**Comment by abin-aot** [comment]
@lmullane @arielleandrews 

- So for the creation of FOI Requests of type "Personal", Are we handling the mandatory documents uploads  like "Proof of guardianship" and other docs for Proxy requests ?. This is an outstanding thing for the current "Save Request" user story as well, because when an analyst change a request type from General to Personal, these mandatory documents need to inputted to the system , right?
-  Assuming the first state of the requests created by the Intake will be "Intake in Progress" ? Because this will directly have an assignee in place while creating..

**Comment by arielleandrews** [comment]
- Guardianship and other consent documents would not be mandatory when creating a request, and this would be the same when switching from General to Personal. The webform is designed with those as mandatory fields to inform applicants of those requirements and have them submit that documentation up front as much as possible. In reality we still see many requests for other's information and we have processes in place to contact applicants and have them submit these after their initial request. Document Uploads will be a separate user story
- Yes once a request has been Created it should be reflected as Intake in Progress (perhaps with a shortcut directly to Open). The example above shows why we would need this - someone might email in a request for their own plus others' information. In that case we would want to Create the request but may need that other documentation from them before we can formally Open the request and start the clock

**Comment by lmullane** [comment]
As discussed, we will create the request which will become 'Intake in Progress." Analyst will have to Open it as they do any request.

**Comment by abin-aot** [comment]
Comments from refinement
 - For Personal requests, Child's FirstName, LastName is NOT mandatory to Capture
- For Personal requests, Another Person's FirstName, LastName is NOT mandatory to capture

**Comment by JHarrietha-AOT** [comment]
Design links have been updated with completed designs for adding a new request

**Comment by m-prodan** [comment]
@lmullane - I am just getting ahead of things, and filling out the QA tests for this story in the master table so we don't have to do it once it's ready for QA. I am thinking we should merge scenario 2 and 3 since they are both regarding mandatory fields?

**Comment by sumathi-thirumani-aot** [comment]
PR : https://github.com/bcgov/foi-flow/pull/320

**Comment by lmullane** [comment]
@sumathi-thirumani-aot, thanks for getting this into TEST.

@m-prodan and I will QA now.

**Comment by lmullane** [comment]
Tested and all the scenarios worked as expected. @m-prodan, let me know how it goes for you?

**Comment by lmullane** [comment]
@sumathi-thirumani-aot, @abin-aot 

Hmmm, was doing some other testing and when I 'Saved' a request submitted online the status became 'Assignment in Progress" instead of "Intake in Progress"? 

What's "Assignment in Progress"?

**Comment by m-prodan** [comment]
@lmullane - I created a personal and general request without issue - both works and are in the queue.

The one item I'll flag is that I noticed some text that persists after the request is saved, giving instructions to only include one piece of contact information. I don't think we need to spell it out like that (and it doesn't match the design), Intake will know they need to enter at least one piece of contact information, so probably best to just remove it.

Also, and this is minor, the design has it as "Add Request" though the AC does say "Create Request" - im agnostic on this one, I think Create Request makes more sense and am happy to keep it as is.

**Comment by lmullane** [comment]
Yeah, the text is on an online request too, even though the info is pre-filled. 

@JHarrietha-AOT, do we need the text message or could we just have a the red boxes around contact criteria, and have them disappear once one thing was filled?

I'm find with 'Add Request.'

**Comment by m-prodan** [comment]
Yes just noticed that now when I went into my online submission.

The first request I submitted online and saved, it went to Intake in Progress as expected. When I went to a request that was already unopened, and saved it - I got the "Assignment in Progress" status:

![image.png](https://images.zenhubusercontent.com/6081a4c82f0c363dddb9c1a0/b7eadaf5-a670-4b49-8839-8ada673cd820)

**Comment by JHarrietha-AOT** [comment]
@lmullane I agree, I don't think the text at the top is enough of a visual indicator of what is missing. I think the message should be in red in the Address and Contact Information Section. And have the missing boxes outlines in red. Once one is completed then the red can disappear.

**Comment by JHarrietha-AOT** [comment]
Also as a note I am getting the toast message saying the request has been successfully saved when it is not, as there is no contact information

**Comment by divyav-aot** [comment]
@lmullane and @JHarrietha-AOT  - We many not be able to highlight the contact boxes as it is either one should be provided not all. That is why given a common message in the top and colored it based on the validation.
Regarding "Assignment in progress" - we will check this issue.

**Comment by divyav-aot** [comment]
@JHarrietha-AOT  - Save button will not be enabled if none of the contact information is given. Is it not like that?

**Comment by JHarrietha-AOT** [comment]
@divyav-aot The save button is enabled if there is not contact information completed, and I click save and get the 'Successfully Saved' Toast. But there is the message at the top saying a need contact information. So it is not actually saved.

**Comment by lmullane** [comment]
@divyav-aot , @JHarrietha-AOT, what if we put a red border around the entire 'ADDRESS AND CONTACT INFORMATION' and removed it when enough of the fields were filled (i.e. email vs street address, postal code, city, country)?

**Comment by JHarrietha-AOT** [comment]
@lmullane I think thats a good solution. We need a visual indicator for users to know the fields that need to be filled out. The text currently doesn't connect enough at the top of the page @divyav-aot

**Comment by lmullane** [comment]
That's the direction I'd like to go. Red around the email and the contact information, and once one is filled we can remove the red border around the other. That would be consistent with the other mandatory fields.

**Comment by divyav-aot** [comment]
@JHarrietha-AOT , @lmullane  - ok. it is possible to give red border for the entire box. What about email? email or address is mandatory right? If we give red border for email and contact box then it is not what we expect right?

**Comment by divyav-aot** [comment]
@JHarrietha-AOT , @lmullane  and @m-prodan  - can you share the issues with steps to reproduce?

**Comment by JHarrietha-AOT** [comment]
@lmullane Exactly red around the fields would be constant with other mandatory fields. A red border around the entire box isn't really constant with other mandatory fields and makes it seem like that section is more important then others - also that the entire section must be completed which isn't true. 

Should we add this under Scenario 3? And Scenario 3 is currently not being met as per my above comments where save is active with no address or email field input. And the success toast pops up but so does message at top. 

@divyav-aot It is happening when you input an address or email in the field and then delete it. So it is not responsive to change.

**Comment by divyav-aot** [comment]
@JHarrietha-AOT  - Thanks Jackie. I will check this issue.
@m-prodan  - can you share the details on "Assignment in progress" state. how to reproduce that?
